### PR TITLE
8346477: Clarify the Java manpage in relation to the JVM's OnOutOfMemoryError flags

### PR DESCRIPTION
**Pull request by dholmes-ora**
Please review this simple clarification to the Java command reference / manpage.

Thanks

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8346477](https://bugs.openjdk.org/browse/JDK-8346477): Clarify the Java manpage in relation to the JVM's OnOutOfMemoryError flags (**Enhancement** - P4)

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**)

### Contributors
 * Thomas Stuefe `<stuefe@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22802/head:pull/22802` \
`$ git checkout pull/22802`

Update a local copy of the PR: \
`$ git checkout pull/22802` \
`$ git pull https://git.openjdk.org/jdk.git pull/22802/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22802`

View PR using the GUI difftool: \
`$ git pr show -t 22802`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22802.diff">https://git.openjdk.org/jdk/pull/22802.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22802#issuecomment-2550157789)
</details>

### CODE REVIEW

**Review by jaikiran — APPROVED** [review]
Hello David, for end users, the distinction between a JVM vs some code running within the Java runtime may not be too clear or obvious. So I think even with this added clarification there might be situations where people will be surprised and will report that these options aren't doing the actions that they expect when OutOfMemoryError is thrown.
But this change is still an improvement to what we currently have. So this looks good to me.

While at it, I noticed in a related JBS issue https://bugs.openjdk.org/browse/JDK-8257790 that you mentioned two additional options `ExitOnOutOfMemory` and `CrashOnOutOfMemoryError`. Those don't seem to be documented in this man page. Are those still relevant and should they be documented too (as a separate issue)?

File: src/java.base/share/man/java.md [hunk 1]
```diff
@@ -1292,10 +1292,13 @@ These `java` options control the runtime behavior of the Java HotSpot VM.
...
 `-XX:OnOutOfMemoryError=`*string*
 :   Sets a custom command or a series of semicolon-separated commands to run
-    when an `OutOfMemoryError` exception is first thrown. If the string
+    when an `OutOfMemoryError` exception is first thrown by the JVM.
+    If the string
     contains spaces, then it must be enclosed in quotation marks. For an
     example of a command string, see the description of the `-XX:OnError`
     option.
+    This does not apply to any `OutOfMemoryError` thrown by Java, or native,
+    code in the JDK libraries.
```
Comment (tstuefe) [inline src/java.base/share/man/java.md:1301]: > This does not apply to any `OutOfMemoryError` thrown by Java, or native, code in the JDK libraries.

This is vague. Does not apply to OOMEs raised explicitly via throw? Or only if thrown from Java code in JDK libraries? So, user code throws are fine? Or, does not apply for OOMEs thrown by JDK libraries regardless of Java or Native? But then, how would a customer know where an exception is thrown without knowing the sources for JDK libraries and hotspot JVM? Why would the throw point even matter?

In my experience, whether or not the OnOOM handling kicks in is a mystery to many customers. We should clearly list the typical points where OnOOM does not work but one would expect it to work, e.g. thread creation errors and explicit throws, regardless of where the throw happens.

**Comment by dholmes-ora** [comment]
Thanks for taking a look at this @jaikiran .

> you mentioned two additional options ExitOnOutOfMemory and CrashOnOutOfMemoryError. Those don't seem to be documented in this man page.

The majority of -XX flags are not documented, and intentionally so. It is somewhat of a legacy situation, whereby -XX flags that tended to be "heavily used" by end users (or were perceived to be potentially useful for end users) were documented. Primarily GC and JIT tuning flags, but also others related to serviceability and debugging. Documenting them can be seen as an encouragement to use them, when that is not the intent for end-users.

File: src/java.base/share/man/java.md (hunk 1 above)
Comment (dholmes-ora) [reply src/java.base/share/man/java.md:1306]: Okay I have revised in the manner suggested. Thanks

**Review by tstuefe — APPROVED** [review]
Thank you for considering my arguments. Looks good and trivial.

File: src/java.base/share/man/java.md (hunk 1 above)
Comment (jaikiran) [reply src/java.base/share/man/java.md:1306]: Hello David, in this latest updated version, this last sentence:

>This does not apply to any `OutOfMemoryError` thrown by Java, or native, code in the JDK libraries.

appears to be a repeat of the newly added longer explanation one line before. Was it intentional keep this line?

**Comment by jaikiran** [comment]
> The majority of -XX flags are not documented, and intentionally so. ... Documenting them can be seen as an encouragement to use them, when that is not the intent for end-users.

Thank you for that detail.

File: src/java.base/share/man/java.md (hunk 1 above)
Comment (dholmes-ora) [reply src/java.base/share/man/java.md]: Ugghh - a mistake. Thanks.

**Review by jaikiran — APPROVED** [review]
The latest update looks good to me.

**Comment by dholmes-ora** [comment]
Thanks for the reviews and suggestions.

**Pull request closed without merging**